### PR TITLE
feat(widget): passive indicator for auto-resuming durable pauses

### PR DESCRIPTION
**Pull request by bookernath**
The unified `await` SSE event can now carry an `awaitReason`. When present, the pause resumes on its own — so the widget shows a passive, non-interactive "working in the background" indicator (new `"pause"` message variant) instead of a resume button, and clears it once the stream continues.

Awaits **without** `awaitReason` (local-tool / WebMCP) keep today's interactive behavior unchanged. Suppression keys on `awaitReason` being present (not a specific value), so future pause kinds work without a release. Indicator copy is overridable via `config.copy.durablePauseLabels`.

### Try it
A new mock-driven demo: **Advanced Examples → Durable Pause** (`apps/web/durable-pause-demo.html`), with a toggle for the two pause kinds. No backend needed.

### Notes
- New unit tests for the event handler and the pause bubble; full widget suite, lint, and typecheck pass.
- Changeset included (minor).

<h3>Greptile Summary</h3>

This PR adds a passive "working in the background" indicator for auto-resuming durable pauses on the unified `await` SSE event. When `awaitReason` is present and non-empty, the widget renders a non-interactive `variant:"pause"` bubble instead of a resume button, and clears it once the stream continues or fails.

- A new `AgentWidgetDurablePause` type and `"pause"` message variant carry the pause state; `resolved` is required (per the prior review thread), and the fingerprint cache is updated so the transition from active → hidden triggers a re-render.
- Settlement is driven by three paths: per-frame early check (any non-ping, non-durable-await frame), `execution_complete`, and a `try/finally` block that clears the bubble on any HTTP/read-layer failure — the latter fixing the network-drop regression flagged in the previous review.
- A mock-driven demo and a unit test suite (including a `pull`-based stream that errors mid-read) cover the happy path, forward-compat unknown reasons, unchanged interactive-await behavior, and the read-layer failure regression.

<h3>Confidence Score: 5/5</h3>

- Safe to merge. The core settlement logic is correct across all exit paths, the prior review's two findings (required `resolved` field, finally-block settle) are both addressed, and the new test suite validates the regression fix directly.
- All three settlement paths (per-frame, execution_complete, stream failure) are exercised by tests and the logic is consistent. The two style-level observations (duplicated guard condition and potential ID collision for sequential pauses in the same execution) are both edge cases with no present misbehavior on the typical single-pause-per-execution path.
- No files require special attention. The two observations in `client.ts` are both in code that handles an uncommon edge (multiple sequential durable pauses in one execution), not the primary path.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/widget/src/client.ts | Core durable-pause logic: adds `durablePauseMessage` state, `settleDurablePause()` helper, per-frame early-settle check (skips `ping`, skips refresh durable-await), and a `try/finally` to clear the bubble on any stream exit. Logic is sound and matches the test suite. |
| packages/widget/src/components/pause-bubble.ts | New passive indicator component. Correctly non-interactive (no buttons/inputs), hides when `resolved`, sets `role="status"` for accessibility, and delegates label resolution to `resolveDurablePauseLabel`. Config override chain is well-ordered. |
| packages/widget/src/types.ts | Adds `AgentWidgetDurablePause` type with required `resolved` field (addressed per prior review thread), new `"pause"` variant, and optional `durablePauseLabels` copy config. Type definitions are clean and well-documented. |
| packages/widget/src/client.test.ts | New test suite covers: known `awaitReason` values, forward-compat unknown reasons, unchanged interactive-tool behavior, stream-resume settlement, and the HTTP read-layer failure regression (using a `pull`-based `ReadableStream` that errors after delivering the await frame). Good coverage. |
| packages/widget/src/utils/message-fingerprint.ts | Adds `awaitReason` and `resolved` to the fingerprint so the cache correctly invalidates when the pause bubble transitions (active → hidden). Both fields are included as distinct segments separated by the `\x00` delimiter. |
| packages/widget/src/utils/__fixtures__/unified-translator.oracle.ts | Propagates `awaitReason`, `crawlId`, and `stepId` through the oracle translator for all three `await`-bearing event types. Additive change; undefined fields are dropped by `JSON.stringify`, so no regression for existing non-durable await frames. |

</details>

<details open><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant Server as Server (SSE)
    participant Client as client.ts (frame loop)
    participant UI as pause-bubble / UI

    Server->>Client: "await {awaitReason:"crawl_pending", ...}"
    Note over Client: durablePauseMessage created<br/>variant:"pause", resolved:false
    Client->>UI: emitMessage(pause, resolved:false)
    UI-->>UI: show passive indicator

    loop keep-alive
        Server->>Client: ping
        Note over Client: payloadType==="ping" → skip early-settle
    end

    Server->>Client: turn_start (stream resumes)
    Note over Client: payloadType!=="ping" & not durable-await<br/>→ settleDurablePause()
    Client->>UI: emitMessage(pause, resolved:true)
    UI-->>UI: hide bubble (display:none)

    Server->>Client: text_delta / text_complete
    Client->>UI: emitMessage(assistantMessage)
    UI-->>UI: render resumed text

    Server->>Client: execution_complete
    Note over Client: finally block → settleDurablePause() (no-op)
```
</details>

<sub>Reviews (3): Last reviewed commit: ["fix: address Greptile findings"](https://github.com/runtypelabs/persona/commit/002745b64523b63f8acd74b12e4628397b359802) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40400525)</sub>

### CODE REVIEW

File: packages/widget/src/types.ts [hunk 1]
```diff
@@ -4422,7 +4430,45 @@ export type AgentWidgetApproval = {
...
+  /** Present for `crawl_pending`: the async-crawl id the server is polling. */
+  crawlId?: string;
+  /** Present for `crawl_pending` and `durable_poll`: the paused step id. */
+  stepId?: string;
+  /**
+   * `false` while the execution is paused; flipped to `true` once the stream
+   * resumes (a subsequent content frame) or the execution completes. The UI
+   * stops the active indicator (and hides the bubble) once resolved.
+   */
+  resolved?: boolean;
```
[the remark this answers is not in the document]
Comment (bookernath) [reply packages/widget/src/types.ts]: Done — made `resolved` required in `dcd84ee`.

File: packages/widget/src/client.ts [hunk 2]
```diff
@@ -1417,6 +1418,26 @@ export class AgentWidgetClient {
...
+    const settleDurablePause = () => {
+      if (!durablePauseMessage || durablePauseMessage.durablePause?.resolved) return;
+      durablePauseMessage.streaming = false;
+      durablePauseMessage.durablePause = {
+        ...(durablePauseMessage.durablePause as AgentWidgetDurablePause),
+        resolved: true,
+      };
+      emitMessage(durablePauseMessage);
+      durablePauseMessage = null;
+    };
```
[the remark this answers is not in the document]
Comment (bookernath) [reply packages/widget/src/client.ts:1574]: Good catch — settled the pause in streamResponse's finally block so it clears on HTTP/read-layer failure, abort, or normal completion. Added a regression test. Fixed in dcd84ee.
